ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -205,30 +211,44 @@ private KsqlEntity executeStatement(
             || statement instanceof DropStream
             || statement instanceof DropTable
     ) {
-      //   getStatementExecutionPlan(statement, statementText, streamsProperties);
+      //Sanity check for the statement before distributing it.
+      validateStatement(statement, statementText, streamsProperties);
       return distributeStatement(statementText, statement, streamsProperties);
     } else {
       if (statement != null) {
-        throw new Exception(String.format(
+        throw new KsqlException(String.format(
             "Cannot handle statement of type '%s'",
             statement.getClass().getSimpleName()
         ));
       } else if (statementText != null) {
-        throw new Exception(String.format(
+        throw new KsqlException(String.format(
             "Unable to execute statement '%s'",
             statementText
         ));
       } else {
-        throw new Exception("Unable to execute statement");
+        throw new KsqlException("Unable to execute statement");
```

Comment:
Can we add the statementText into the exception so we know what weren't able to execute?

Reply:
Done.